test_tablets.py::test_tablet_missing_data_repair fails due to `NoHostAvailable`

```
cassandra.cluster.NoHostAvailable: ('Unable to complete the operation against any hosts', {<Host: 127.157.135.26:9042 datacenter1>: ConnectionException('Pool for 127.157.135.26:9042 is shutdown')})
...
    @pytest.mark.repair
    @pytest.mark.asyncio
    async def test_tablet_missing_data_repair(manager: ManagerClient):
...
        for idx in range(0,3):
            s = servers[idx].server_id
            await manager.server_stop_gracefully(s, timeout=120)
>           await check()

test/topology_experimental_raft/test_tablets.py:482: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    async def check():
        logger.info("Checking table")
        query = SimpleStatement("SELECT * FROM test.test;", consistency_level=ConsistencyLevel.ONE)
>       rows = await cql.run_async(query)
E       cassandra.cluster.NoHostAvailable: ('Unable to complete the operation against any hosts', {<Host: 127.157.135.26:9042 datacenter1>: ConnectionException('Pool for 127.157.135.26:9042 is shutdown')})

```

see https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/6229/

[scylla-1370.log](https://github.com/scylladb/scylladb/files/14125397/scylla-1370.log)
[topology_experimental_raft.test_tablets.4.log](https://github.com/scylladb/scylladb/files/14125398/topology_experimental_raft.test_tablets.4.log)

--- a/test/pylib/manager_client.py
+++ b/test/pylib/manager_client.py
@@ -10,12 +10,13 @@
 """
 
 from concurrent.futures import ThreadPoolExecutor
-from typing import List, Optional, Callable, Any
+from typing import List, Optional, Callable, Any, Awaitable
 from time import time
 import logging
 from test.pylib.log_browsing import ScyllaLogFile
 from test.pylib.rest_client import UnixRESTClient, ScyllaRESTAPIClient, ScyllaMetricsClient
 from test.pylib.util import wait_for
+from test.pylib.util import wait_for_cql_and_get_hosts
 from test.pylib.internal_types import ServerNum, IPAddress, HostID, ServerInfo
 from test.pylib.scylla_cluster import ReplaceConfig, ScyllaServer
 from cassandra.cluster import Session as CassandraSession  # type: ignore # pylint: disable=no-name-in-module
@@ -165,7 +166,7 @@ class ManagerClient():
         await self.server_sees_others(server_id, wait_others, interval = wait_interval)
         self._driver_update()
 
-    async def rolling_restart(self, servers):
+    async def rolling_restart(self, servers: List[ServerInfo], with_down: Optional[Callable[[ServerInfo], Awaitable[Any]]] = None):
         for idx, s in enumerate(servers):
             await self.server_stop_gracefully(s.server_id)
 
@@ -175,6 +176,11 @@ class ManagerClient():
             for idx2 in range(len(servers)):
                 if idx2 != idx:
                     await self.server_not_sees_other_server(servers[idx2].ip_addr, s.ip_addr)
+
+            if with_down:
+                up_servers = [u for u in servers if u.server_id != s.server_id]
+                await wait_for_cql_and_get_hosts(self.cql, up_servers, time() + 60)
+                await with_down(s)
 
             await self.server_start(s.server_id)
 

--- a/test/pylib/manager_client.py
+++ b/test/pylib/manager_client.py
@@ -193,6 +193,8 @@ class ManagerClient():
                 if idx2 != idx:
                     await self.server_sees_other_server(servers[idx2].ip_addr, s.ip_addr)
 
+        await wait_for_cql_and_get_hosts(self.cql, servers, time() + 60)
+
     async def server_pause(self, server_id: ServerNum) -> None:
         """Pause the specified server."""
         logger.debug("ManagerClient pausing %s", server_id)

--- a/test/topology_experimental_raft/test_tablets.py
+++ b/test/topology_experimental_raft/test_tablets.py
@@ -453,22 +453,20 @@ async def test_tablet_missing_data_repair(manager: ManagerClient):
     await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
 
     keys_list = [range(0, 100), range(100, 200), range(200, 300)]
+    keys_for_server = dict([(s.server_id, keys_list[idx]) for idx, s in enumerate(servers)])
     keys = range(0, 300)
 
-    for idx in range(0,3):
-        s = servers[idx].server_id
-        await manager.server_stop_gracefully(s, timeout=120)
-        logger.info(f"Stopped server {idx}");
-        logger.info(f"Insert into server {idx}");
-        await asyncio.gather(*[cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({k}, {k});") for k in keys_list[idx]])
-        await manager.server_start(s)
-        logger.info(f"Started server {idx}");
+    async def insert_with_down(down_server):
+        logger.info(f"Stopped server {down_server.server_id}")
+        logger.info(f"Insert into server {down_server.server_id}")
+        await asyncio.gather(*[cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({k}, {k});")
+                               for k in keys_for_server[down_server.server_id]])
 
-    await wait_for_cql_and_get_hosts(cql, servers, time.time() + 60)
+    await manager.rolling_restart(servers, with_down=insert_with_down)
 
     await repair_on_node(manager, servers[0], servers)
 
-    async def check():
+    async def check_with_down(down_node):
         logger.info("Checking table")
         query = SimpleStatement("SELECT * FROM test.test;", consistency_level=ConsistencyLevel.ONE)
         rows = await cql.run_async(query)
@@ -476,13 +474,8 @@ async def test_tablet_missing_data_repair(manager: ManagerClient):
         for r in rows:
             assert r.c == r.pk
 
-    for idx in range(0,3):
-        s = servers[idx].server_id
-        await manager.server_stop_gracefully(s, timeout=120)
-        await check()
-        await manager.server_start(s)
+    await manager.rolling_restart(servers, with_down=check_with_down)
 
-    await cql.run_async("DROP KEYSPACE test;")
 
 @pytest.mark.asyncio
 async def test_tablet_cleanup(manager: ManagerClient):

